lib/core/src/persist/sync.rs

```diff
@@ -0,0 +1,163 @@
+use anyhow::Result;
+use rusqlite::{named_params, params, Row};
+
+use crate::sync::model::{sync::Record, SyncDetails};
+
+use super::Persister;
+
+impl Persister {
+    pub(crate) fn get_latest_revision(&self) -> Result<i64> {
+        let con = self.get_connection()?;
+
+        let latest_revision: i64 = con
```

Comment:
I think this mechanism is not reliable as it counts on the fact that synced items are sent (and received) at the same order without skipped items. It doesn't have to be the case as the server may handle concurrent updates which trigger events but there is not gurantee they will be received at the same order.
If we call list_changes then we can persist the latest revision as the last one and count on it.